### PR TITLE
Adding Arabic to localization

### PR DESCRIPTION
**Pull request by OdaiHatemMOE**
Fixes #

## Changelog Entry

Adding  Arabic  localization 

## Description

Adding  Arabic  localization 

## Specific Changes

Adding ar-AR.js to localization 

---

-  [ ] Testing Added

### CODE REVIEW

File: packages/component/src/Localization/ar-AR.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        return `الامس`;
+    }
+    return getLocaleString(date, 'ar-AR');
+}
+
+function botSaidSomething(avatarInitials, text) {
+    return `Bot ${avatarInitials} said, ${text}`;
+}
+
+function downloadFileWithFileSize(downloadFileText, fileName, size) {
```
[the remark this answers is not in the document]
Comment (OdaiHatemMOE) [reply packages/component/src/Localization/ar-AR.js]: Done

File: packages/component/src/Localization/ar-AR.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    // Full text should read: "Download file <filename> of size <filesize>"
+    return `${downloadFileText} ${fileName} of size ${size}`;
+}
+
+function uploadFileWithFileSize(fileName, size) {
```
[the remark this answers is not in the document]
Comment (OdaiHatemMOE) [reply packages/component/src/Localization/ar-AR.js]: Done

File: packages/component/src/Localization/ar-AR.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    return `${fileName} of size ${size}`;
+}
+
+function userSaidSomething(avatarInitials, text) {
```
[the remark this answers is not in the document]
Comment (OdaiHatemMOE) [reply packages/component/src/Localization/ar-AR.js]: Done

File: packages/component/src/Localization/ar-AR.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    return `User ${avatarInitials} said, ${text}`;
+}
+
+export default {
+    CONNECTED_NOTIFICATION: 'متصل',
+    FAILED_CONNECTION_NOTIFICATION: 'غير قادر على الاتصال.',
+    INITIAL_CONNECTION_NOTIFICATION: 'اتصال ...',
+    INTERRUPTED_CONNECTION_NOTIFICATION: 'حدث انقطاع في الشبكة. إعادة اتصال ... ',
+    RENDER_ERROR_NOTIFICATION: 'Render error. Please check the console or contact the bot developer.',
```
[the remark this answers is not in the document]
Comment (OdaiHatemMOE) [reply packages/component/src/Localization/ar-AR.js]: Done

File: packages/component/src/Localization/ar-AR.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    // Do not localize {Retry}; it is a placeholder for "Retry". English translation should be, "Send failed. Retry."
+    SEND_FAILED_KEY: `فشل إرسال. {Retry}.`,
+    SLOW_CONNECTION_NOTIFICATION: 'يستغرق وقتا أطول من المعتاد للاتصال.',
+    'Bot said something': botSaidSomething,
+    'User said something': userSaidSomething,
+    'X minutes ago': xMinutesAgo,
+    // '[File of type '%1']': '[File of type '%1']",
+    // '[Unknown Card '%1']': '[Unknown Card '%1']',
+    'Adaptive Card parse error': 'Adaptive Card parse error',
```
[the remark this answers is not in the document]
Comment (OdaiHatemMOE) [reply packages/component/src/Localization/ar-AR.js]: Done

File: packages/component/src/Localization/ar-AR.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    'Adaptive Card render error': 'Adaptive Card render error',
```
[the remark this answers is not in the document]
Comment (OdaiHatemMOE) [reply packages/component/src/Localization/ar-AR.js]: Done

File: packages/component/src/Localization/ar-AR.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    SentAt: 'ارسلت ب',
+    Speak: 'تكلم',
+    'Starting…': 'بداية ... ',
+    Tax: 'ضريبة',
+    Total: 'كلياً',
+    'Type your message': 'اكتب سؤالك',
+    TypingIndicator: 'عرض مؤشر الكتابة',
+    'Upload file': 'تحميل ملف',
+    UploadFileWithFileSize: uploadFileWithFileSize,
+    UserSent: 'ارسل المستخدم:',
```
[the remark this answers is not in the document]
Comment (OdaiHatemMOE) [reply packages/component/src/Localization/ar-AR.js]: Done

File: packages/component/src/Localization/ar-AR.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    'Listening…': 'الاستماع ...',
+    'New messages': 'رسائل جديدة',
+    Retry: 'حاول مرة اخرى',
+    Right: 'يمين',
+    Send: 'ارسل',
+    Sending: 'ارسال',
+    SendStatus: 'حالة الارسال: ',
+    SentAt: 'ارسلت ب',
+    Speak: 'تكلم',
+    'Starting…': 'بداية ... ',
```
[the remark this answers is not in the document]
Comment (OdaiHatemMOE) [reply packages/component/src/Localization/ar-AR.js]: Done

**Comment by corinagum** [comment]
It looks like the tests are failing because of linting errors. Could you run `npm run eslint` at the root of Web Chat and correct those errors? Please let me know if you have questions.

**Comment by corinagum** [comment]
@OdaiHatemMOE gentle ping, would you be able to take a look at the linting errors? Thank you!

**Comment by corinagum** [comment]
This PR is stale, so I am closing this issue. We will implement this in the future **with original credit**, and tracking can be found at #1976

**Pull request closed without merging**